test/test_energy_storage_device.py

```diff
@@ -132,5 +134,121 @@ def test_pycap_simulation(self):
                                    delta=1e-5)
             self.assertAlmostEqual(o1.current[i], o2.current[i])
 
+    def test_accuracy_pycap_simulation(self):
+        #
+        # tests the accuracy of a pycap simulation against a 
+        # straight run dualfoil sim with different timesteps
+        #
+        df1 = Dualfoil(path=path)  # will use pycap
+        df2 = Dualfoil(path=path)  # manual runs
+        im = df_manip.InputManager(path=path)
+
+        # testing a charge-to-hold-const-voltage
+        # manual 
+        # use InputManager to set the input file
+        c = -10.0  # constant charge current
+        # charge for 4 minutes straight
+        im.add_new_leg(c, 4.0, 1)
+        df1.run()
+        df1.outbot.update_output()
+        v = 4.54  # constant voltage
+        # hold constant voltage for 3 minutes straight
+        im.add_new_leg(v, 3.0, 0)
+        df1.run()
+        df1.outbot.update_output()
+
+        # pycap simulation
+        # build a ptree input
+        ptree = PropertyTree()
+        ptree.put_double('time_step', 20.0)  # 20 second time step
+        ptree.put_string('charge_mode', 'constant_current')
+        ptree.put_double('charge_current', 10.0)
+        ptree.put_string('charge_stop_at_1', 'voltage_greater_than')
+        ptree.put_double('charge_voltage_limit', 4.54)
+        ptree.put_bool('charge_voltage_finish', True)
+        # hold end voltage after either 3 minutes have passed
+        # OR current falls under 1 ampere
+        ptree.put_double('charge_voltage_finish_max_time', 180.0)
+        ptree.put_double('charge_voltage_finish_current_limit', 1.0)
+
+        const_current_const_voltage = Charge(ptree)
+        const_current_const_voltage.run(df2)
+
+        o1 = df1.outbot
+        o2 = df2.outbot
+        o2.write_main_output()
+
+        # affirm we make it this far and have usable data
+        self.assertTrue(len(o1.time) > 0)
+        self.assertTrue(len(o2.time) > 0)
+        # lengths of data should be different
+        self.assertFalse(len(o1.time) == len(o2.time))
+
+        # Create one combined list of tuples with form 
+        #         (time, current, voltage).
+        # Compare the consistency of the two simulation ouputs
+        # by checking smooth changes within the curves of the 
+        # combined output lists
+        output = []
+        for i in range(len(o1.time)):
+            tuple = (o1.time[i], o1.current[i], o1.potential[i])
```

Comment:
`tuple` is a Python data type just like `list` and it is bad practice to use it as a variable name and assign it some value.